lib/gui.py

```diff
@@ -96,18 +96,101 @@ def on_submit():
     submit_button.grid(row=3, column=1)
 
 
+def make_port_scan(main_window):
+    font = ""
+    width = 800
+    height = 600
+    port_scan_window = tk.Toplevel(main_window, bg="#222", width=width, height=height)
+    port_scan_window.resizable(False, False)
+    port_scan_window.title("Port Scanner")
+
+    port_scan_frame = tk.Frame(port_scan_window)
+    port_scan_frame.pack(fill=tk.BOTH, expand=True)
+
+    host = tk.StringVar()
+    host.set("127.0.0.1")
+    port_range = tk.StringVar()
+    port_range.set("1-65535")
+    threads = tk.StringVar()
+    threads.set(multiprocessing.cpu_count())
+
+    scan_label = tk.Label(port_scan_frame, text="Select scanning options", bg="#222", fg="#FFF")
+    scan_label.pack(fill=tk.X, expand=True)
+
+    def validate_input(host_address, ports_range, n_threads):
+        host_address = host_address.replace("default: ", "")
+        ports_range = ports_range.replace("default: ", "")
+        n_threads = n_threads.replace("default: ", "")
+        import re
+
+        regex = "^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$"
```

Comment:
`regex` -> `ipv4_regex`. I found a simpler one: `^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}$`

Reply:
But it doesn't work

![image](https://github.com/PixelToolkit/PixelToolkit/assets/113286903/a16e12d8-b915-47be-a430-8cbd681652a2)

Reply:
either stackoverflow is lying or something is not right: https://stackoverflow.com/questions/5284147/validating-ipv4-addresses-with-regexp

Reply:
If it works, let it be

Reply:
It didn't work because of string escaping in python, this is the corrected regex: `^((25[0-5]|(2[0-4]|1\\d|[1-9]|)\\d)\\.?\\b){4}$`

Reply:
btw, does it matter if the regex is 10 characters longer? I mean, can any of us fix it without stack or chatgpt if there happens to be error in regex syntax anyway?

Reply:
I like to think I understand regex. But then it breaks and I realize I don't understand regex. But neither does chatgpt. Now always anyway. Ahh it can stay as it is I guess

Reply:
btw there is a `ipaddress` library which can be used to validate ipv4 / ipv6 addresses like so:
```python
def is_ipv4_valid(ip):
    try:
        ipaddress.IPv4Address(ip)
        return True
    except ipaddress.AddressValueError:
        return False
```

Reply:
let's stay with that regex for now